File: apps/ResidentApp.iOS/Sources/ComplaintDTOs.swift
import Foundation

// MARK: - Nested DTOs

struct MediaDTO: Codable, Identifiable {
    let mediaId: Int
    let url: String
    let type: String
    let uploadedAt: Date

    var id: Int { mediaId }
}

struct WorkNoteDTO: Codable, Identifiable {
    let workNoteId: Int
    let content: String
    let staffMemberId: Int
    let staffMemberName: String
    let createdAt: Date

    var id: Int { workNoteId }
}

struct StaffMemberSummaryDTO: Codable, Identifiable {
    let staffMemberId: Int
    let fullName: String
    let jobTitle: String?
    let availability: StaffState

    var id: Int { staffMemberId }
}

// MARK: - ComplaintDTO

struct ComplaintDTO: Codable, Identifiable {
    let complaintId: Int
    let propertyId: Int
    let unitId: Int
    let unitNumber: String?
    let buildingName: String?
    let residentId: Int
    let residentName: String?
    let title: String
    let description: String
    let category: String
    let urgency: Urgency
    let status: TicketStatus
    let permissionToEnter: Bool
    let assignedStaffMember: StaffMemberSummaryDTO?
    let media: [MediaDTO]
    let workNotes: [WorkNoteDTO]
    let eta: Date?
    let createdAt: Date
    let updatedAt: Date
    let resolvedAt: Date?
    let tat: Double?
    let residentRating: Int?
    let residentFeedbackComment: String?

    var id: Int { complaintId }
}

// MARK: - Paginated Response

struct ComplaintsPageDTO: Decodable {
    let items: [ComplaintDTO]
    let totalCount: Int
    let page: Int
    let pageSize: Int
}

// MARK: - Request DTOs

struct TriggerSosRequest: Encodable {
    let title: String
    let description: String
    let permissionToEnter: Bool
}

struct SubmitFeedbackRequest: Encodable {
    let rating: Int
    let comment: String?
}

File: apps/StaffApp.Android/app/src/main/java/com/acls/staff/data/remote/dto/ComplaintDtos.kt
package com.acls.staff.data.remote.dto

import com.google.gson.annotations.SerializedName

data class MediaDto(
    @SerializedName("mediaId") val mediaId: Int,
    @SerializedName("url") val url: String,
    @SerializedName("type") val type: String,
    @SerializedName("uploadedAt") val uploadedAt: String
)

data class WorkNoteDto(
    @SerializedName("workNoteId") val workNoteId: Int,
    @SerializedName("content") val content: String,
    @SerializedName("staffMemberId") val staffMemberId: Int,
    @SerializedName("staffMemberName") val staffMemberName: String?,
    @SerializedName("createdAt") val createdAt: String
)

data class StaffMemberSummaryDto(
    @SerializedName("staffMemberId") val staffMemberId: Int,
    @SerializedName("fullName") val fullName: String,
    @SerializedName("jobTitle") val jobTitle: String?,
    @SerializedName("availability") val availability: String?
)

data class ComplaintDto(
    @SerializedName("complaintId") val complaintId: Int,
    @SerializedName("propertyId") val propertyId: Int,
    @SerializedName("unitId") val unitId: Int,
    @SerializedName("unitNumber") val unitNumber: String?,
    @SerializedName("buildingName") val buildingName: String?,
    @SerializedName("residentId") val residentId: Int,
    @SerializedName("residentName") val residentName: String?,
    @SerializedName("title") val title: String,
    @SerializedName("description") val description: String,
    @SerializedName("category") val category: String,
    @SerializedName("urgency") val urgency: String,
    @SerializedName("status") val status: String,
    @SerializedName("permissionToEnter") val permissionToEnter: Boolean,
    @SerializedName("assignedStaffMember") val assignedStaffMember: StaffMemberSummaryDto?,
    @SerializedName("media") val media: List<MediaDto>?,
    @SerializedName("workNotes") val workNotes: List<WorkNoteDto>?,
    @SerializedName("eta") val eta: String?,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updatedAt") val updatedAt: String,
    @SerializedName("resolvedAt") val resolvedAt: String?,
    @SerializedName("tat") val tat: Double?,
    @SerializedName("residentRating") val residentRating: Int?,
    @SerializedName("residentFeedbackComment") val residentFeedbackComment: String?
)

data class ComplaintsPageDto(
    @SerializedName("items") val items: List<ComplaintDto>,
    @SerializedName("totalCount") val totalCount: Int,
    @SerializedName("page") val page: Int,
    @SerializedName("pageSize") val pageSize: Int
)

data class AddWorkNoteRequestDto(
    @SerializedName("content") val content: String
)

data class UpdateComplaintStatusRequestDto(
    @SerializedName("status") val status: String
)

data class UpdateEtaRequestDto(
    @SerializedName("eta") val eta: String
)
